test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -118,21 +149,34 @@ session_simulator::get_read_timestamp() const
     return _read_ts;
 }
 
+bool
+session_simulator::get_ts_round_prepared()
+{
+    return _ts_round_prepared;
+}
+
 bool
 session_simulator::get_ts_round_read()
 {
     return _ts_round_read;
 }
 
-void
-session_simulator::set_commit_timestamp(uint64_t ts)
+int
+session_simulator::set_commit_timestamp(uint64_t commit_ts)
 {
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+    WT_SIM_RET(ts_manager->validate_commit_timestamp(this, commit_ts));
     if (!_has_commit_ts) {
-        _first_commit_ts = ts;
+        _first_commit_ts = commit_ts;
         _has_commit_ts = true;
     }
 
-    _commit_ts = ts;
+    if (_ts_round_prepared && commit_ts < _prepare_ts)
+        _commit_ts = _prepare_ts;
+    else if (commit_ts >= _prepare_ts)
+        _commit_ts = commit_ts;
```

Comment:
We also set durable when we set commit.

Reply:
Yeah I've handled this in the next durable ticket.